src/games/supported/DemonAttack.cpp

```diff
@@ -121,3 +129,41 @@ void DemonAttackSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect DemonAttackSettings::getAvailableModes(){
+    ModeVect modes;
+    modes.push_back(1);
+    modes.push_back(3);
+    modes.push_back(5);
+    modes.push_back(7);
+
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void DemonAttackSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+    if(m==1 || m==3 || m==5 || m==7){
```

Comment:
My point is only to be careful with the c++11 features we do decide to use,
as support can be spotty on OSX.

On Tue, Mar 29, 2016 at 5:19 PM, alcinos <notifications@github.com> wrote:

> In src/games/supported/DemonAttack.cpp
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/115#discussion_r57810773>
> :
>
> > @@ -121,3 +129,41 @@ void DemonAttackSettings::loadState(Deserializer & ser) {
> >    m_lives = ser.getInt();
> >  }
> >
> > +// returns a list of mode that the game can be played in
> > +ModeVect DemonAttackSettings::getAvailableModes(){
> > +    ModeVect modes;
> > +    modes.push_back(1);
> > +    modes.push_back(3);
> > +    modes.push_back(5);
> > +    modes.push_back(7);
> > +
> > +    return modes;
> > +}
> > +
> > +// set the mode of the game
> > +// the given mode must be one returned by the previous function
> > +void DemonAttackSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
> > +    if(m==1 || m==3 || m==5 || m==7){
>
> @mhauskn <https://github.com/mhauskn> I don't really see your point since
> we don't use (and AFAICT don't plan to) use c++11 threading capabilities
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/115/files/3e0f85d22e9df239aa4e17374b524efcfe59aa37#r57810773>
>